test/support/counting_predicates.hpp

```diff
@@ -0,0 +1,59 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef TEST_SUPPORT_COUNTING_PREDICATES_H
+#define TEST_SUPPORT_COUNTING_PREDICATES_H
+
+#include <oneapi/dpl/cstddef>
+
+template <typename Predicate, typename Arg>
+struct unary_counting_predicate {
+public:
+    typedef Arg argument_type;
+    typedef bool result_type;
+
+    unary_counting_predicate(Predicate p) : p_(p), count_(0) {}
+    ~unary_counting_predicate() {}
+
+    bool operator () (const Arg &a) const { ++count_; return p_(a); }
+    dpl::size_t count() const { return count_; }
+    void reset() { count_ = 0; }
+
+private:
+    Predicate p_;
+    mutable dpl::size_t count_;
+};
+
+
+template <typename Predicate, typename Arg1, typename Arg2=Arg1>
+struct binary_counting_predicate {
```

Comment:
I have "grep"ed all of the files in this PR and seems like `binary_counting_predicate` is not used.
So I guess this code should be removed.
`unary_counting_predicate` is only used in `ref_2.pass.cpp`, so I would consider moving the implementation of `unary_counting_predicate` into this `.cpp` file if it would not be used in other PRs with tests integration.

Reply:
Probably this structure may be used in some another PR's with tests. I guess we should not delete it now.

Reply:
I have checked all other parts of tests integration and `binary_counting_predicate` is not used there. 
So I suggest not to introduce dead code inside of the repo and remove this unnecessary structure.

Reply:
Fixed